ambry-router/src/main/java/com.github.ambry.router/DecryptJob.java

```diff
@@ -70,27 +87,33 @@ public void run() {
    * @param gse the {@link GeneralSecurityException} that needs to be set while invoking callback for the job
    */
   public void closeJob(GeneralSecurityException gse) {
-    callback.onCompletion(new DecryptJobResult(blobId, null), gse);
+    callback.onCompletion(null, gse);
   }
 
   /**
    * Class respresenting decrypt job result
    */
   class DecryptJobResult {
     private final BlobId blobId;
-    private final ByteBuffer decryptedContent;
+    private final ByteBuffer decryptedBlobContent;
+    private final ByteBuffer decryptedUserMetadata;
 
-    DecryptJobResult(BlobId blobId, ByteBuffer decryptedContent) {
+    DecryptJobResult(BlobId blobId, ByteBuffer decryptedBlobContent, ByteBuffer decryptedUserMetadata) {
       this.blobId = blobId;
-      this.decryptedContent = decryptedContent;
+      this.decryptedBlobContent = decryptedBlobContent;
+      this.decryptedUserMetadata = decryptedUserMetadata;
     }
 
     BlobId getBlobId() {
```

Comment:
if `blobId` is not necessary, let us omit this.

Reply:
Not really a blocker. Will take it up in next patch.